code/game/machinery/telecomms/machines/hub.dm

```diff
@@ -34,6 +34,36 @@
 
 	use_power(idle_power_usage)
 
+/obj/machinery/telecomms/hub/update_power()
+	var/old_on = on
+	if(toggled)
+		if(machine_stat & (BROKEN|NOPOWER|EMPED)) // if powered, on. if not powered, off. if too damaged, off
+			on = FALSE
+			soundloop.stop()
+		else
+			on = TRUE
+			soundloop.start()
+	else
+		on = FALSE
+		soundloop.stop()
+	if(old_on != on)
+		update_appearance()
+
+/obj/machinery/telecomms/hub/Destroy()
+	GLOB.telecomms_list -= src
+	QDEL_NULL(soundloop)
+	for(var/obj/machinery/telecomms/comm in GLOB.telecomms_list)
+		remove_link(comm)
+	links = list()
+	return ..()
+
+/obj/machinery/telecomms/hub/Initialize(mapload)
+	. = ..()
+	soundloop = new(src, on)
+	GLOB.telecomms_list += src
+	if(mapload && autolinkers.len)
+		return INITIALIZE_HINT_LATELOAD
```

Comment:
where is LateInitialize defined here?

Reply:
It's in ``code\game\machinery\telecomms\telecomunications.dm`` I think?

Reply:
ok but why are we doing this here when we werent before?

Reply:
Because it's how the other telecomms machines work.

Reply:
Then it will be transferred with `. = ..()` and you don't need to include it yourself

Reply:
so just return ``.=..()``

Reply:
So I can just remove ``INITIALIZE_HINT_LATELOAD``

Reply:
Any code that is just from the base of the telecomms machine does not need to be in here because `..()` does the same thing.

Same for `Destroy`. I'm now realizing I'm skeptical the `telecomms_list += src` is doing anything `..()` isn't (other than potentially breaking some logic)

Reply:
I followed your advice here and it works just fine now, sans the extra additional code. I'd appreciate if you could review it.

I did a test in-game, the hub works fine, the sound stops when it needs to, and then starts back up.